src/libraries/System.Linq.Expressions/src/System/Dynamic/DynamicObject.cs

```diff
@@ -97,7 +98,7 @@ public virtual bool TryInvokeMember(InvokeMemberBinder binder, object[] args, ou
         /// <param name="result">The result of the conversion.</param>
         /// <returns>true if the operation is complete, false if the call site should determine behavior.</returns>
         [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1007:UseGenericsWhereAppropriate")]
-        public virtual bool TryConvert(ConvertBinder binder, out object result)
+        public virtual bool TryConvert(ConvertBinder binder, out object? result)
```

Comment:
>out [](start = 61, length = 3)

Should `result` be marked `[NotNullWhen(true)]`?

Reply:
From the implementation code seems it can return null with `true` return. no?

Reply:
Yes, you're right, the implementation should be able to set the result to `null` and return `true`.

---
In reply to: [362650799](https://github.com/dotnet/runtime/pull/1064#discussion_r362650799) [](ancestors = 362650799)